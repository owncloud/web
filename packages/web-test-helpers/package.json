{
  "name": "@ownclouders/web-test-helpers",
  "version": "11.3.2",
  "description": "ownCloud web test helpers",
  "license": "AGPL-3.0",
  "private": false,
  "type": "module",
  "main": "src/index.ts",
  "author": "ownCloud GmbH <devops@owncloud.com>",
  "homepage": "https://github.com/owncloud/web/tree/master/packages/web-test-helpers",
  "repository": {
    "type": "git",
    "url": "https://github.com/owncloud/web",
    "directory": "packages/web-test-helpers"
  },
  "files": [
    "dist/"
  ],
  "publishConfig": {
    "directory": "package",
    "linkDirectory": false,
    "exports": {
      ".": {
        "import": "./dist/web-test-helpers.es.js",
        "require": "./dist/web-test-helpers.umd.cjs",
        "types": "./dist/src/index.d.ts"
      }
    }
  },
  "scripts": {
    "vite": "vite",
    "prepublishOnly": "rm -rf ./package && clean-publish && find package && cat package/package.json",
    "postpublish": "rm -rf ./package"
  },
  "peerDependencies": {
    "@ownclouders/web-pkg": "workspace:^",
    "@vue/test-utils": "^2.4.6",
    "vue": "^3.5.10"
  },
  "devDependencies": {
    "@vitejs/plugin-vue": "5.1.4",
    "clean-publish": "^5.0.0",
    "vite-plugin-dts": "^4.2.3",
    "vite": "^5.4.8"
  },
  "dependencies": {
    "@casl/ability": "^6.7.1",
    "@casl/vue": "^2.2.2",
    "@ownclouders/design-system": "workspace:^",
    "@ownclouders/web-client": "workspace:^",
    "@pinia/testing": "^0.1.3",
    "axios": "1.7.7",
    "pinia": "2.2.6",
    "vitest-mock-extended": "2.0.2",
    "vue-router": "4.2.5",
    "vue3-gettext": "2.4.0"
  }
}
