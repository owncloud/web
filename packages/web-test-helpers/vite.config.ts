import { resolve } from 'path'
import { defineConfig } from 'vite'
import dts from 'vite-plugin-dts'
import pkg from './package.json' assert { type: 'json' }
import vue from '@vitejs/plugin-vue'

export default defineConfig({
  build: {
    lib: {
      entry: resolve(__dirname, 'src/index.ts'),
      name: 'web-test-helpers',
      fileName: (format) => `web-test-helpers.${format}.js`
    },
    rollupOptions: {
      external: [...Object.keys(pkg.dependencies), ...Object.keys(pkg.peerDependencies)]
    }
  },
  plugins: [vue(), dts({ include: ['src'], outDir: 'dist/types', insertTypesEntry: true })]
})
