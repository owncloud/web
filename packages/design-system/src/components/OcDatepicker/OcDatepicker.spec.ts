import Datepicker from './OcDatepicker.vue'
import { ComponentProps, defaultPlugins, shallowMount } from 'web-test-helpers'
import { DateTime } from 'luxon'
import { nextTick } from 'vue'

describe('OcDatePicker', () => {
  it('renders', () => {
    const wrapper = getWrapper({ label: 'Datepicker label' })
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('sets the initial date correctly', async () => {
    const wrapper = getWrapper({ label: 'Datepicker label', currentDate: DateTime.now() })
    await nextTick()
    const inputEl = wrapper.find('.oc-text-input').element as HTMLInputElement
    expect(inputEl.value).toEqual(DateTime.now().toISODate())
  })
  it('sets the minimum date correctly', async () => {
    const wrapper = getWrapper({ label: 'Datepicker label', minDate: DateTime.now() })
    await nextTick()
    const inputEl = wrapper.find('.oc-text-input')
    expect(inputEl.attributes('min')).toEqual(DateTime.now().toISODate())
  })
  it('emits event on date change', async () => {
    const wrapper = getWrapper({ label: 'Datepicker label' })
    const inputEl = wrapper.find('.oc-text-input')
    await inputEl.setValue(DateTime.now().toISODate())
    expect(wrapper.emitted('dateChanged')).toBeTruthy()
  })
})

function getWrapper(props: ComponentProps<typeof Datepicker>) {
  return shallowMount(Datepicker, {
    props,
    global: {
      plugins: [...defaultPlugins()],
      stubs: {
        OcTextInput: false
      }
    }
  })
}
