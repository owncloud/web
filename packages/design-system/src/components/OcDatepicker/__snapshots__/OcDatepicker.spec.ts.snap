// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OcDatePicker > renders 1`] = `
"<div class="oc-date-picker"><label class="oc-label" for="oc-textinput-1">Datepicker label</label>
  <div class="oc-position-relative">
    <!--v-if--> <input id="oc-textinput-1" aria-invalid="false" class="oc-text-input oc-input oc-rounded" type="date">
    <!--v-if-->
  </div>
  <div class="oc-text-input-message">
    <!--v-if--> <span id="oc-textinput-1-message" class=""></span>
  </div>
  <!--v-if-->
</div>"
`;
